### PR TITLE
Merge PropertiesI.java and Properties.java

### PR DESCRIPTION
**Pull request by externl**
This PR merges PropertiesI and Properties in Java

### CODE REVIEW

**Review by bernardnormier — APPROVED** [review]

File: java/src/Ice/src/main/java/com/zeroc/Ice/Properties.java [hunk 1]
```diff
@@ -1,34 +1,158 @@
...
+
 /**
  * A property set used to configure Ice and Ice applications. Properties are key/value pairs, with
  * both keys and values being strings. By convention, property keys should have the form
  * <em>application-name</em>[.<em>category</em>[.<em>sub-category</em>]].<em>name</em>.
  */
-public interface Properties {
+public final class Properties {
+  static class PropertyValue {
+    public PropertyValue(PropertyValue v) {
```
Comment (pepone) [inline java/src/Ice/src/main/java/com/zeroc/Ice/Properties.java:16]: Do we need to keep this constructor, or just `clone` like in C#?

File: java/src/Ice/src/main/java/com/zeroc/Ice/Util.java [hunk 2]
```diff
@@ -12,9 +12,10 @@ public final class Util {
    * Creates a new empty property set.
    *
    * @return A new empty property set.
+   * @deprecated Use {@link Properties#Properties()} instead.
    */
   public static Properties createProperties() {
```
Comment (pepone) [inline java/src/Ice/src/main/java/com/zeroc/Ice/Util.java:17]: Missing `@Deprecated`?

**Review by pepone — APPROVED** [review]
Looks good, see comments.

File: java/src/Ice/src/main/java/com/zeroc/Ice/Properties.java (hunk 1 above)
Comment (externl) [reply java/src/Ice/src/main/java/com/zeroc/Ice/Properties.java:16]: I meant to remove it and forgot!

**Pull request merged**